src/users/users.service.ts

```diff
@@ -156,31 +155,67 @@ export class UsersService {
           `${studentUser.email} - konto jest nieaktywne`,
         );
       }
-      if (student.status !== StudentStatus.Available) {
-        throw new BadRequestException(
-          `Użytkownik ${studentUser.email} nie jest dostępny`,
-        );
-      }
-      student.status = newStatus;
-      await this.studentProfileRepository.save(student);
 
-      if (student.status === StudentStatus.Hired) {
+      if (newStatus === StudentStatus.Hired) {
+        if (student.status !== StudentStatus.DuringRecruitment) {
+          throw new BadRequestException(
+            `Użytkownik ${studentUser.email} nie jest przez ciebie zarezerwowany.`,
+          );
+        }
+
+        student.status = newStatus;
+        const reservedStudent = await this.reservedStudentsRepository.findOne({
+          where: {
+            student: {
+              user: {
+                id: studentId,
+              },
+            },
+          },
+        });
+        if (reservedStudent) {
+          await this.studentProfileRepository.save(student);
+          await this.reservedStudentsRepository.delete(reservedStudent.id);
```

Comment:
```suggestion
          await this.reservedStudentsRepository.delete(reservedStudent.id);
```
dlaczego ten delete tutaj skoro jest w metodzie 'changeStatusStatusToHired ?